library/src/main/java/com/northconcepts/templatemaster/form/CrudResource.java

```diff
@@ -162,26 +164,30 @@ public String getEditBodyTemplate() {
     
     @GET
     @Path("/")
-    public Response getListRecords(@QueryParam("q") String searchQuery, @QueryParam("s")String sortField) throws Throwable {
-        return getListRecords(searchQuery, sortField, 0);
+    public Response getListRecords(@QueryParam("q") String searchQuery, @QueryParam("s")String sortField, @QueryParam("p") Integer pageSize) throws Throwable {
+        return getListRecords(searchQuery, sortField, 0, pageSize);
     }
     
     @GET
     @Path("/{pageNumber}")
-    public Response getListRecords(@QueryParam("q") String searchQuery, @QueryParam("s")String sortField, @PathParam("pageNumber") int pageNumber) {
+    public Response getListRecords(@QueryParam("q") String searchQuery, @QueryParam("s")String sortField, @PathParam("pageNumber") int pageNumber, @QueryParam("p") Integer pageSize) {
         if (pageNumber == 0 && Util.isEmpty(sortField) && Util.isNotEmpty(formDef.getDefaultSortField())) {
             Url url = RequestHolder.getUrl().setQueryParam("s", formDef.getDefaultSortField());
             return gotoUri(url.toString());
         }
         
-        searchQuery =  Util.isNotEmpty(searchQuery)?searchQuery:null;
-        sortField = Util.isNotEmpty(sortField)?sortField:null;
+        if (pageSize != null && pageSize != 0 && !Util.isValidPageSize(pageSize)) {
```

Comment:
I think this should be: `if (pageSize == null || !Util.isValidPageSize(pageSize)) {`

Reply:
We have refactored the code.